scripts/generate-favicons.ts

```diff
@@ -0,0 +1,65 @@
+import { favicons, config as faviconsConfig } from 'favicons';
+import config from '../src/data/config';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const fsPromises = fs.promises;
+const faviconsDirectory = './public/favicons';
+
+const deleteFile = async (file: string) => {
+  await fsPromises.unlink(file);
+  console.log(`${file} has been deleted successfully`);
+};
+
+const removeFaviconFiles = async (folderPath: string) => {
+  const files = await fsPromises.readdir(folderPath);
+
+  for (const file of files) {
+    await deleteFile(path.join(folderPath, file));
+  }
+};
+
+(async () => {
+  const { faviconPath } = config.meta;
+
+  const response = await favicons(`.${faviconPath}`, {
+    ...faviconsConfig.defaults,
+    path: '/favicons',
+    appName: config.meta.title,
+    appDescription: config.meta.description,
+    appShortName: config.meta.title,
+    lang: config.i18n.locale.code,
+    start_url: '.',
+    icons: {
+      android: ['android-chrome-192x192.png', 'android-chrome-512x512.png'],
+      windows: false,
+      yandex: false,
+      appleStartup: false,
+      appleIcon: ['apple-touch-icon.png'],
+      favicons: ['favicon-16x16.png', 'favicon-32x32.png', 'favicon.ico'],
+    },
+  });
+
+  if (!fs.existsSync(faviconsDirectory)) {
+    await fs.promises.mkdir(faviconsDirectory);
+  }
+
+  await removeFaviconFiles(faviconsDirectory);
+
+  for (const file of [...response.images, ...response.files]) {
+    await fsPromises.writeFile(`${faviconsDirectory}/${file.name}`, file.contents);
+    console.log(`${file.name} has been created successfully`);
+  }
```

Comment:
Nie musisz tego dodawać bo to już mikro-optymalizacja, ale generalnie wydajniej byłoby wynieść zawartość tego for'a do funkcji i zrobić.
```
await Promise.all([...response.images, ...response.files].map(saveFile));
```